Assets/BossRoom/Scripts/Shared/Game/State/GameStateBehaviour.cs

```diff
@@ -94,18 +94,18 @@ protected virtual void OnDestroy()
 
         protected virtual void OnApplicationQuit()
         {
-            if (!isActiveAndEnabled)
+            if (!isActiveAndEnabled || NetworkManager.Singleton)
```

Comment:
Do we still need to be a NetworkBehaviour?

Reply:
Probably yes, as the subclasses have some data that needs to be replicated/ RPC'ed